components/post_view/post_recent_reactions/post_recent_reactions.tsx

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {Tooltip} from 'react-bootstrap';
+import {Dispatch} from 'redux';
+
+import store from 'stores/redux_store.jsx';
+import {getCurrentLocale} from 'selectors/i18n';
+import {getEmojiMap} from 'selectors/emojis';
+import Permissions from 'mattermost-redux/constants/permissions';
+import {Emoji} from 'mattermost-redux/types/emojis';
+import {Locations} from 'utils/constants';
+import ChannelPermissionGate from 'components/permissions_gates/channel_permission_gate';
+import OverlayTrigger from 'components/overlay_trigger';
+
+import EmojiItem from './recent_reactions_emoji_item';
+
+type LocationTypes = 'CENTER' | 'RHS_ROOT' | 'RHS_COMMENT';
+
+type Props = {
+    channelId?: string;
+    postId: string;
+    teamId: string;
+    getDotMenuRef: () => HTMLDivElement;
+    location: LocationTypes;
+    emojis: Emoji[];
+    actions: {
+        addReaction: (postId: string, emojiName: string) => (dispatch: Dispatch) => void;
+    };
+}
+
+type State = {
+    location: LocationTypes;
+    showEmojiPicker: boolean;
+}
+
+const state = store.getState();
+const emojiMap = getEmojiMap(state);
```

Comment:
This too would belong in a `mapStateToProps` function in the component's index file.